src/gasPriceOracle/oracle.ts

```diff
@@ -24,17 +24,17 @@ async function legacy(provider: providers.Provider, chainId: number): Promise<Ga
   };
 }
 
-// @todo: Update to ethers 5.7.x to access FeeData.lastBaseFeePerGas
+// @todo: Update to ethers 5.7.x to access FeeData.lastBaseFeePerGas. Use feeData.gasPrice until then.
 // https://github.com/ethers-io/ethers.js/commit/8314236143a300ae81c1dcc27a7a36640df22061
 async function eip1559(provider: providers.Provider, chainId: number): Promise<GasPriceEstimate> {
   const feeData: providers.FeeData = await provider.getFeeData();
 
-  [feeData.maxFeePerGas, feeData.maxPriorityFeePerGas].forEach((field: BigNumber | null) => {
+  [feeData.gasPrice, feeData.maxPriorityFeePerGas].forEach((field: BigNumber | null) => {
     if (!BigNumber.isBigNumber(field) || field.lt(0)) error("getFeeData()", chainId, feeData);
   });
 
   const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas as BigNumber;
-  const maxFeePerGas = feeData.maxFeePerGas as BigNumber;
+  const maxFeePerGas = feeData.gasPrice as BigNumber; // note gasPrice is used.
```

Comment:
Before submitting this, I want to confirm whether we should be adding maxPriorityFeePerGas here.

Reply:
Judging from [this implementation](https://github.com/ethers-io/ethers.js/blob/c80fcddf50a9023486e9f9acb1848aba4c19f7b6/packages/abstract-provider/src.ts/index.ts#L235-L256), what we ultimately want our estimate to be is: `lastBaseFeePerGas + maxPriorityFeePerGas`. If `lastBaseFeePerGas` is being reported as `gasPrice` (I think this is what you're saying), then we want the caller to estimate their gas costs by doing: `gasPrice + maxPriorityFeePerGas`.

I don't think the caller should ever use maxFeePerGas to _estimate_ their costs, because maxFeePerGas is a huge overestimate (2x base fee ish). If this is what we're currently doing, then maybe we should sub `gasPrice` for maxFeePerGas as you're doing now, but ultimately, we should change the naming because maxFeePerGas isn't what we're intending to use to feed the gas estimate.

Does that make sense?

Reply:
> Does that make sense?

Yep - all good. I was thrown off because I'd implemented maxFeePerGas = gasPrice + maxPriorityFeePerGas, and suddenly the tests failed unexpectedly. I've fixed that now.